src/Lecture3.hs

```diff
@@ -174,11 +191,18 @@ together only different elements.
 [4,2,2]
 >>> appendDiff3 [1 .. 5] [1 .. 5] [1 .. 5]
 [1,2,3,4,5]
+
 >>> appendDiff3 (Product 2) (Product 3) (Product 3)
 Product {getProduct = 6}
 
 -}
-appendDiff3 = error "TODO"
+appendDiff3 :: (AppendDiffeable a) => a -> a -> a -> a
+appendDiff3 = ad3
+
+class (Semigroup a) => AppendDiffeable a where
+  ad3 :: a -> a -> a -> a
+
+-- Q1: adding a typeclass like this makes sense?
```

Comment:
There's no need to add such a typeclass. You need to add the `Eq` constraint to the `appendDiff3` function and avoid adding equal elements 👍🏻

Reply:
thanks!